website/project/views/comment.py

```diff
@@ -270,31 +463,79 @@ def undelete_comment(**kwargs):
     return {}
 
 
-@must_be_logged_in
-@must_be_contributor_or_public
-def update_comments_timestamp(auth, **kwargs):
-    node = kwargs['node'] or kwargs['project']
+def _update_comments_timestamp(auth, node, page='node', root_id=None):
+    if node.is_contributor(auth.user) and page != 'total':
+        user_timestamp = auth.user.comments_viewed_timestamp
+        if not user_timestamp.get(node._id, None):
+            user_timestamp[node._id] = dict()
+        node_timestamp = user_timestamp.get(node._id, None)
+        if node_timestamp and isinstance(node_timestamp, datetime):
```

Comment:
It looks like @huangginny is checking for any `user.comments_viewed_timestamp` fields that are still equal to a timestamp instead of a dictionary of timestamps for each page. This shouldn't be necessary after the migration -- should I take it out @sloria?